boot/zephyr/CMakeLists.txt

```diff
@@ -252,6 +252,26 @@ if(CONFIG_MCUBOOT_SERIAL)
     )
 endif()
 
+if(CONFIG_SOC_ESP32 OR CONFIG_SOC_ESP32S2 OR CONFIG_SOC_ESP32C3 OR CONFIG_SOC_ESP32S3)
+
+  if(CONFIG_SOC_ESP32)
+    zephyr_compile_definitions(CONFIG_ESP32_XTAL_FREQ=40)
+  endif()
+  zephyr_compile_definitions(BOOTLOADER_BUILD=1)
+  zephyr_compile_definitions(CONFIG_MCUBOOT=1)
+  zephyr_compile_definitions(NDEBUG=1)
+
+  zephyr_include_directories(${BOOT_DIR}/espressif/include)
+  zephyr_include_directories(${BOOT_DIR}/espressif/hal/include)
+
+  zephyr_sources (
+    ${BOOT_DIR}/espressif/port/esp_loader.c
+    ${BOOT_DIR}/espressif/hal/src/bootloader_init_common.c
+    ${BOOT_DIR}/espressif/hal/src/bootloader_wdt.c
+    ${BOOT_DIR}/espressif/hal/src/${CONFIG_SOC}/bootloader_init.c
+  )
```

Comment:
As convenient as it may be, it seems conceptually wrong for one port (`boot/zephyr`) to reference files from any other port (`boot/espressif`).

Reply:
First I used `hal_espressif` versions of `boot/espressif/hal/src/bootloader_.*.c` files, but any change/fix in the `bootloader_init` function asked to expose port private/static members which ends up colliding/creating multiple definitions due to IDF sources organization. Then I noticed very same sources are available in the `boot/espressif/` which lacks all the disadvantages and solves all previous issues. The `esp_loader.c` reference is different and much easier to remove from the Zephyr port. 

I feel your concerns, but think about this: It is wrong to reference the code from different SOC port just to use some function I am lazy to write. But here we share the same SOC early init functions (loaded by the the same ROM loader), then continuing entirely Zephyr way. So I think it is close to DRY principle.

But, maybe there are other reasons for not doing it and I'm open to discuss :)

Reply:
> I feel your concerns, but think about this: It is wrong to reference the code from different SOC port just to use some function I am lazy to write. But here we share the same SOC early init functions (loaded by the the same ROM loader), then continuing entirely Zephyr way. So I think it is close to DRY principle.

Applying DRY principle here comes at the cost of introducing coupling between two distinct ports, which in my opinion makes it prohibitive. So for now I believe duplicating the code is a better alternative.